ipa-core/src/report.rs

```diff
@@ -413,6 +434,186 @@ where
     }
 }
 
+/// A binary report as submitted by a report collector, containing encrypted `OprfReport`
+/// An `EncryptedOprfReport` consists of:
+///     `ct_mk`: Enc(`match_key`)
+///     `ct_btt`: Enc(`breakdown_key`, `trigger_value`, `timestamp`)
+///     associated data of `ct_mk`: `key_id`, `epoch`, `event_type`, `site_domain`,
+
+#[derive(Copy, Clone, Eq, PartialEq)]
+pub struct EncryptedOprfReport<BK, TV, TS, B>
+where
+    B: Deref<Target = [u8]>,
+    BK: SharedValue,
+    TV: SharedValue,
+    TS: SharedValue,
+{
+    data: B,
+    phantom_data: PhantomData<(BK, TV, TS)>,
+}
+
+/// follows the outline of the implementation of `EncryptedReport`
+impl<B, BK, TV, TS> EncryptedOprfReport<BK, TV, TS, B>
+where
+    B: Deref<Target = [u8]>,
+    BK: SharedValue,
+    TV: SharedValue,
+    TS: SharedValue,
+    Replicated<BK>: Serializable,
+    Replicated<TV>: Serializable,
+    Replicated<TS>: Serializable,
+{
+    const ENCAP_KEY_MK_OFFSET: usize = 0;
+    const CIPHERTEXT_MK_OFFSET: usize = Self::ENCAP_KEY_MK_OFFSET + EncapsulationSize::USIZE;
+    // need to round up ciphertext length to nearest multiple of 16 since AES has block length 16
+    const ENCAP_KEY_BTT_OFFSET: usize = (Self::CIPHERTEXT_MK_OFFSET
+        + U16::USIZE
+        + ((<Replicated<BA64> as Serializable>::Size::USIZE) + 15))
```

Comment:
I'd suggest formalizing the assumption that the match key is `BA64` by defining something like `pub type OprfIpaMatchKey = BA64`.

Reply:
I agree we should consider different length match keys, especially a BA128. I would prefer to add this as a generic like TS, BK or TV rather than a pub type. I have added an issue, see: https://github.com/private-attribution/ipa/issues/916